### PR TITLE
Entrant email lists

### PR DESCRIPTION
**Pull request by jmacdotorg**
This depends on #134.

Adds a new admin feature that simply displays two linebreak-separated email lists, ready for pasting into a tool like Mailman.

There is a list of unique email addresses of authors associated with the current competition, and an "anti-list" of addresses of authors whose entries exist but are all disqualified.

The idea being: After intents close, the organizer can create a mailing list from the first email address list. And after the final deadline passes and the organizer has DQ's all entries that didn't make it across the line, the organizer can mass-remove from the mailing list all the addresses from the "anti-list".

### CODE REVIEW

File: IFComp/lib/IFComp/Controller/Admin/Email.pm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            'entries.is_disqualified' => $is_disqualified,
+        },
+        {   join     => 'entries',
+            group_by => 'email',
+            order_by => 'email',
+        },
+    )->get_column('email')->all;
+
+    return @emails;
+    die "Got this: @emails";
```
Comment (adherzog) [inline IFComp/lib/IFComp/Controller/Admin/Email.pm]: Why?

File: IFComp/root/src/admin/email/index.tt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+</div>
+[% ELSE %]
+<p>...actually, it looks like we don't have any email addresses right now.</p>
+[% END %]
+
+[% IF anti_emails.size > 0 %]
+<h2>The anti-list</h2>
+<p>Furthermore, here is a list of all the email addresses for authors with <em>only</em> disqualified entries for [% current_comp.year %]. You can use this list to mass-unsubscribe addresses from a mailing list you may have made earlier based on pre-deadline intents.</p>
+<p>
+<div class="well">
```
Comment (adherzog) [inline IFComp/root/src/admin/email/index.tt]: The list above this one in the template has the `<p>` inside the `<div>`, but this one has the `<div>` inside the `<p>`.

File: IFComp/lib/IFComp/Controller/Admin/Email.pm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    } @disqualified_emails;
+
+    $c->stash(
+        emails      => \@emails,
+        anti_emails => \@anti_emails,
+    );
+
+}
+
+sub _get_emails {
```
Comment (adherzog) [inline IFComp/lib/IFComp/Controller/Admin/Email.pm]: Please move this method into the schema, along with the calculation of the anti_emails above. Ideally the controller will call a method (or two) on some result/resultset to retrieve the two lists, and then just put the lists into the stash for the template. The controller shouldn't have to do any direct query writing or list manipulation.

**Comment by jmacdotorg** [comment]
Updated with the requested changes, but holding off on adding tests until #136 is merged. That introduces some refactoring to IFCompTest.pm that I'd like to take advantage of here.

**Comment by adherzog** [comment]
Please move the `emails_for_comp()` and `anti_emails_for_comp()` methods out of the schema class and into the comp object.

**Comment by jmacdotorg** [comment]
This is ready for a fresh check.

**Comment by adherzog** [comment]
Needs tests.

**Comment by jmacdotorg** [comment]
After using this branch for production-information purposes I have found some clear flaws and shortcomings with it, so I'm withdrawing this pull request for further tooling.

**Pull request closed without merging**